src/main/java/net/masterthought/cucumber/reducers/ReportFeatureWithRetestMerger.java

```diff
@@ -0,0 +1,103 @@
+package net.masterthought.cucumber.reducers;
+
+import net.masterthought.cucumber.json.Element;
+import net.masterthought.cucumber.json.Feature;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Merge list of given features. If there are couple of scenarios with the same Id then
+ * only the latest will be stored into the report.
+ *
+ * Uses when need to generate a report with rerun results of failed tests.
+ */
+final class ReportFeatureWithRetestMerger implements ReportFeatureMerger {
+
+    private static final String ERROR = "You are not able to use this type of results merge. The start_timestamp field" +
+            " should be part of element object. Please, update the cucumber-jvm version.";
+    private static final ElementComparator ELEMENT_COMPARATOR = new ElementComparator();
+
+    @Override
+    public List<Feature> merge(List<Feature> features) {
+        Map<String, Feature> mergedFeatures = new HashMap<>();
+        for (Feature candidate : features) {
+            Feature mergedFeature = mergedFeatures.get(candidate.getId());
+            if (mergedFeature == null) {
+                mergedFeatures.put(candidate.getId(), candidate);
+            }
+            else {
+                updateElements(mergedFeatures.get(candidate.getId()), candidate.getElements());
+            }
+        }
+        return new ArrayList<>(mergedFeatures.values());
+    }
+
+    private void updateElements(Feature feature, Element[] elements) {
+        for (int i = 0; i < elements.length; i++) {
+            Element current = elements[i];
+            if (current.isScenario()) {
+                checkArgument(current.getStartTime() != null, ERROR);
```

Comment:
I don't understand how this should work with report generated by new and old cucumber

Reply:
OK so my concerns are:
- If this is an error shouldn't we throw an exception?
- This will be reported for each scenario so I expect it will be never printed or all the time what makes a little mess
- This is nice message for debug purpose but for sure most of users won't read logs :)

Reply:
@dabogee and this